connect/src/spirc.rs

```diff
@@ -490,7 +493,22 @@ impl SpircTask {
                 },
                 connection_id_update = self.connection_id_update.next() => match connection_id_update {
                     Some(result) => match result {
-                        Ok(connection_id) => self.handle_connection_id_update(connection_id),
+                        Ok(connection_id) => {
+                            self.handle_connection_id_update(connection_id);
+
+                            // pre-acquire access_token, preventing multiple request while running
```

Comment:
This comment sees out of place, now that you've moved the acquisition to line 341, correct?

Reply:
No, here we per-acquire the access-token. On line 341 we per-acquire the client-token. This comment additionally explains why we acquire the access-token here and not earlier.

Reply:
OK!